src/Users/Utente.java

```diff
@@ -78,5 +81,13 @@ public Roles getRole() {
         return role;
     }
 
+    public Carrello getCarrelloCliente() {
 
+        if(carrelloCliente == null) carrelloCliente = new Carrello();
```

Comment:
Rimane il fatto che i get non dovrebbero contenere logica. Se non è troppo dispendioso farei in modo che il get ritorni null e questo caso venga eventualmente gestito da chi chiama il metodo, considerandolo come il caso in cui il carrello è vuoto.

Reply:
Ok, ora il getter ritorna solo il valore, la gestione del valore null è stata affidata al metodo che lo richiama.